packages/components/src/List/ListComposition/DisplayMode/DisplayMode.component.js

```diff
@@ -0,0 +1,103 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Navbar, NavDropdown, Nav, MenuItem } from 'react-bootstrap';
+import uuid from 'uuid';
+
+import { useListContext } from '../context';
+import Icon from '../../../Icon';
+
+function getIcon(selected) {
+	switch (selected) {
+		case 'table':
+			return 'talend-table';
+		case 'large':
+			return 'talend-expanded';
+		default:
+			return 'talend-table';
+	}
+}
+
+function getLabel(option, t) {
+	switch (option) {
+		case 'table':
+			return t('LIST_SELECT_DISPLAY_MODE_TABLE', { defaultValue: 'Table' });
+		case 'large':
+			return t('LIST_SELECT_DISPLAY_MODE_LARGE', { defaultValue: 'Expanded' });
+		default:
+			return option;
+	}
+}
+
+function DisplayMode(props) {
+	const { displayMode, setDisplayMode, t } = useListContext();
+	const {
+		id,
+		displayModes,
+		initialDisplayMode,
+		onChange,
+		selectedDisplayMode = displayMode,
+	} = props;
+
+	useEffect(() => {
```

Comment:
Is this useEffect handle some kind of componentDidMount stuff ?
My main concern is why do we trigger it on onChange modification ?

Reply:
You're right, it's not supposed to support controlled/uncontrolled switch.
And if this changes, it will reset the state. In all cases it's not right. I remove the onChange from dependency to just execute the effect once.